### PR TITLE
first preview

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Command/ExtractCommand.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        $this->extractor = $extractor;
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Extracts translator IDs from files.')
+            ->addOption('languages', 'L', InputOption::VALUE_REQUIRED, 'Creating translations for languages. By default only `en`.', 'en')
+            ->addOption('category', 'C', InputOption::VALUE_REQUIRED, 'Default category for unsetted category messages.', $this->defaultCategory)
```
Comment (samdark) [inline src/Command/ExtractCommand.php:36]: Is it really required? I see it being set to `app` already.

File: src/Command/ExtractCommand.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        self::$defaultName = 'translator/extract';
+        $this->extractor = $extractor;
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Extracts translator IDs from files.')
+            ->addOption('languages', 'L', InputOption::VALUE_REQUIRED, 'Creating translations for languages. By default only `en`.', 'en')
```
Comment (samdark) [inline src/Command/ExtractCommand.php:35]:
```suggestion
            ->addOption('languages', 'L', InputOption::VALUE_REQUIRED, 'Comma separated list of languages to write message sources for. By default it is `en`.', 'en')
```

File: src/Command/ExtractCommand.php (hunk 1 above)
Comment (samdark) [inline src/Command/ExtractCommand.php:36]:
```suggestion
            ->addOption('category', 'C', InputOption::VALUE_REQUIRED, 'Default message category to use when category is not set.', $this->defaultCategory)
```

File: src/Command/ExtractCommand.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            ->addOption('category', 'C', InputOption::VALUE_REQUIRED, 'Default category for unsetted category messages.', $this->defaultCategory)
+            ->addOption('except', 'E', InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL, 'Exclude path from extracting.', [])
+            ->addOption('only', 'O', InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL, 'Only path from extracting.', [])
```
Comment (samdark) [inline src/Command/ExtractCommand.php:38]:
```suggestion
            ->addOption('only', 'O', InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL, 'Use the only specified path for extracting.', [])
```

File: src/Command/ExtractCommand.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            ->addArgument('path', InputArgument::OPTIONAL, 'Path for extracts message IDs.')
```
Comment (samdark) [inline src/Command/ExtractCommand.php:39]:
```suggestion
            ->addArgument('path', InputArgument::OPTIONAL, 'Path for extracting message IDs.')
```

File: src/Command/ExtractCommand.php (hunk 1 above)
Comment (darkdef) [reply src/Command/ExtractCommand.php]: User may be wants extract message from another module, with changed translation category (for example with use  method `withCategory`)

Comment (samdark) [reply src/Command/ExtractCommand.php]: Yes but why making the option required?

Comment (darkdef) [reply src/Command/ExtractCommand.php]: It'w wrong - fixed. But all Option in symfony/console optional

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Awesome!

**Pull request merged**